application/app/services/dataverse/dataset_service.rb

```diff
@@ -0,0 +1,14 @@
+module Dataverse
+  class DatasetService
+    include LoggingCommon
+    include DateTimeCommon
+
+    class UnauthorizedException < Exception; end
+
+    def initialize(dataverse_url, http_client: Common::HttpClient.new(base_url: dataverse_url), file_utils: Common::FileUtils.new)
+      @dataverse_url = dataverse_url
+      @http_client = http_client
+      @file_utils = file_utils
+    end
+  end
+end
```

Comment:
I will develop this class in future PRs. The idea is to split the existing Service class into many classes